src/client/components/HealthCareApp.jsx

```diff
@@ -74,10 +76,38 @@ class HealthCareApp extends React.Component {
     this.scrollToTop();
   }
 
-  handleSubmit() {
+  handleSubmit(e) {
+    e.preventDefault();
     const path = this.props.location.pathname;
-    this.context.store.dispatch(updateSubmissionStatus('applicationSubmitted'));
-    this.context.store.dispatch(updateCompletedStatus(path));
+    const store = this.context.store;
+    const veteran = store.getState().veteran;
+
+    store.dispatch(updateSubmissionStatus('submitPending'));
+    store.dispatch(updateCompletedStatus(path));
+
+    // Pretty-print the data that we're about to POST
+    console.log(JSON.stringify(veteran, null, 2));
+
+    // POST data to endpoint
+    fetch('/hca/', {
+      method: 'POST',
+      header: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      timeout: 10000, // 10 seconds
+      body: JSON.stringify(veteran)
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      return response;
+    }).then(response => {
```

Comment:
Why chain so much in this promise?  There's nothing async happening after the first response...  I feel like you could just call `store.dispatch(updateSubmissionStatus('submitFailed', error));` in line 102 and `store.dispatch(updateSubmissionStatus('submitSucceeded', response.json()));` instead of returning response in 104.